org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/backend/SonarLintEclipseClient.java

```diff
@@ -334,9 +336,36 @@ public void showSoonUnsupportedMessage(ShowSoonUnsupportedMessageParams params)
     });
   }
 
+  /**
+   *  Before this method is even invoked, some magic already interacts with Eclipse: When there is no connection to
+   *  SonarQube, it will automatically launch the "Connect to SonarQube or SonarCloud" wizard. Some magic will also ask
+   *  the user to find the correct project if there is currently no binding to the SonarQube project.
+   *  -> both cases must not be tested afterwards
+   */
   @Override
   public void showIssue(ShowIssueParams params) {
-    // Not yet implemented
+    var configScopeId = params.getConfigScopeId();
+
+    // We were just asked to find the correct project, this cannot happen -> Only log the information
+    var projectOpt = resolveProject(configScopeId);
+    if (projectOpt.isEmpty()) {
+      SonarLintLogger.get().error("Open in IDE: The project '" + configScopeId
+        + "' was removed in the middle of running the action on '" + params + "'");
+      return;
+    }
+    var project = projectOpt.get();
+    
+    // We were just asked to connect and create a binding, this cannot happen -> Only log the information
+    var bindingOpt = SonarLintCorePlugin.getServersManager().resolveBinding(project);
+    if (bindingOpt.isEmpty()) {
+      SonarLintLogger.get().error("Open in IDE: The project '" + configScopeId
+        + "' removed its binding in the middle of running the action on '" + params + "'");
+      return;
+    }
+    
+    // Handle expensive checks and actual logic in separate job to not block the thread
+    new OpenIssueInEclipseJob(new OpenIssueInEclipseJobParams("Open in IDE", params, project, bindingOpt.get()))
```

Comment:
Before using `OpenIssueInEclipseJobParams` I had multiple constructors on `OpenIssueInEclipseJob` (some of them were private because only invoked internally).
When I changed it to ask the user to change his preferences (and therefore would invoke the job after pressing **Apply and Close**) I needed something to be easily handed over to the `SonarLintPreferencePage` and I didn't want to save all the information one-by-one but rather as some form of structure.

I see the point, do you have some alternative in mind?

Reply:
Then maybe rename the class to something like `OpenIssueContext`